text/0059-otlp-trace-data-format.md

```diff
@@ -0,0 +1,348 @@
+  // attributes. If this value is 0, then no attributes were dropped.
+  int32 dropped_attributes_count = 11;
+
+  // TimedEvent is a time-stamped annotation of the span, consisting of either
+  // user-supplied key-value pairs, or details of a message sent/received between Spans.
+  message TimedEvent {
+    // time_unixnano is the time the event occurred.
+    int64 time_unixnano = 1;
+
+    // name is a user-supplied description of the event.
+    string name = 2;
+
+    // attributes is a collection of attribute key/value pairs on the event.
+    repeated AttributeKeyValue attributes = 3;
+
+    // dropped_attributes_count is the number of dropped attributes. If the value is 0,
+    // then no attributes were dropped.
+    int32 dropped_attributes_count = 4;
+  }
+
+  // timed_events is a collection of TimedEvent items.
+  repeated TimedEvent timed_events = 12;
+
+  // dropped_timed_events_count is the number of dropped timed events. If the value is 0,
+  // then no events were dropped.
+  int32 dropped_timed_events_count = 13;
+
+  // Link is a pointer from the current span to another span in the same trace or in a
+  // different trace. For example, this can be used in batching operations,
+  // where a single batch handler processes multiple requests from different
+  // traces or when the handler receives a request from a different project.
+  // See also Links specification:
+  // https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/overview.md#links-between-spans
+  message Link {
+    // trace_id is a unique identifier of a trace that this linked span is part of.
+    // The ID is a 16-byte array.
+    bytes trace_id = 1;
+
+    // span_id is a unique identifier for the linked span. The ID is an 8-byte array.
+    bytes span_id = 2;
+
+    // tracestate is the trace state associated with the link.
+    repeated TraceStateEntry tracestate = 3;
+
+    // attributes is a collection of attribute key/value pairs on the link.
+    repeated AttributeKeyValue attributes = 4;
+
+    // dropped_attributes_count is the number of dropped attributes. If the value is 0,
+    // then no attributes were dropped.
+    int32 dropped_attributes_count = 5;
+  }
+
+  // links is a collection of Links, which are references from this span to a span
+  // in the same or different trace.
+  repeated Link links = 14;
+
+  // dropped_links_count is the number of dropped links after the maximum size was
+  // enforced. If this value is 0, then no links were dropped.
+  int32 dropped_links_count = 15;
+
+  // status is an optional final status for this span. Semantically when status
+  // wasn't set it is means span ended without errors and assume Status.Ok (code = 0).
+  Status status = 16;
+
+  // child_span_count is an optional number of child spans that were generated while this
+  // span was active. If set, allows an implementation to detect missing child spans.
```

Comment:
this is unclear, per resolved thread below. Is this the number of local-only children? If so, the doc string should be explicit about it.

Reply:
Added clarification to make it explictely about local children.